hyperspy/misc/math_tools.py

```diff
@@ -69,3 +69,42 @@ def anyfloatin(things):
         if isfloat(n) and not n.is_integer():
             return True
     return False
+
+
+def outer_nd(*vec):
+    """
+    Calculates outer product of n vectors
+
+    Parameters
+    ----------
+    vec : vector
+
+    Return
+    ------
+    out : ndarray
+    """
+    return reduce(np.multiply.outer, vec)
+
+
+def hann_window_nth_order(m, order):
+    """
+    Calculates 1D Hann window of nth order
+
+    Parameter
+    ---------
+    m : int
+        number of points in window
+    order : int
+        Filter order
+
+    Return
+    ------
+    window : array
+        window
+    """
+    sin_arg = np.pi * (m - 1.) / m
+    cos_arg = 2. * np.pi / (m - 1.) * (np.arange(m))
+    window = 0.
```

Comment:
Just a suggestion, feel free to ignore if you think it makes things unclear:
You can do away with the window = 0 and the loop by using
`window = sum([(-1) ** i / i * np.sin(i * sin_arg) * (np.cos(i * cos_arg) - 1) for i in range(1, order+1)])`